packages/gatsby/package.json

```diff
@@ -201,17 +214,20 @@
     "graphql": "^14.6.0"
   },
   "scripts": {
-    "build": "npm run build:src && npm run build:internal-plugins && npm run build:rawfiles && npm run build:cjs",
+    "build": "npm run build:src && npm run build:internal-plugins && npm run build:rawfiles && npm run build:cjs && npm run build:recipes",
```

Comment:
I'd argue this shouldn't be in `packages/gatsby`. Why not a new package for all the recipes like `gatsby-recipes`? Every new recipe would now need a gatsby upgrade. Eg if we update docs with available recipes, we'd have to specify when they were added.

Making actual recipes be part of `gatsby-cli` would also be reasonable.

Reply:
yeah definitely makes sense — we'll do that today

Reply:
Reminder about that, this is a critical blocker without fix of which this won't get merged.

Reply:
From my point of view it makes sense to create gatsby-recipe packages that holds all possible commands, graphql server, ... so this can easily be added as a gatsby-cli dependency and to gatsby-desktop without much fuzz.

Reply:
@johno is in progress of moving the actual recipes to a `gatsby-recipes` package — as for the rest of the code — I could go either way — what Desktop/Admin need is the ability to spin up a GraphQL server that lets them read/write to the Gatsby site — having that part of the core `gatsby` package makes sense as it is core `gatsby` functionality & not a standalone thing — Desktop will just reach into the site to start the GraphQL server.

Reply:
This sounds brittle - providers are inside Gatsby, means versions of gatsby-recipes won't be compatible with some version of Gatsby. In addition this all isn't really Gatsby functionality, but more stuff to operate on Gatsby :D Maybe `gatsby-recipes` and `gatsby-recipes-api`? `gatsby-recipes` can define what version of `gatsby-recipes-api` they depend on.

Reply:
I'm following @freiksenet here.

**my train of thought:**
Recipes will be used on multiple platforms, gatsby-cli, cloud?, gatsby admin? gatsby desktop? Multiple consumers mean shared code. They all need to touch providers. I don't see any connection with gatsby core as recipes provides its own graphql server and all. Unsure what gatsby-admin will be, but it's probably going to be an internal plugin that hooks into the express middleware to add the admin route, this plugin can tell recipes to run its graphql server. The same goes for all other platforms.

Why not core? Well, adding this to core means people will need to upgrade gatsby each time to run the latest version. Gatsby-cli isn't directly coupled to gatsby, so people won't be able to run gatsby recipes outside of a gatsby dir, so scaffolding a new project won't be possible at all.  How will we show proper error messages like `recipe not found, please upgrade gatsby`.

Reply:
Do recipes actually need Gatsby? They seem like something that would have value for any web project? Can they become a generic IAC system for the frontend?

Reply:
Technically not but it makes sense to brand it with the gatsby cli. That's also what we want the refactor to be like.

Thanks for the extra 👍